#104 Add export functionality

JosefDvorak opened the pull request:
This pull request adds a new UI to allow selecting and running modules in bulk. The results are saved in xlsx, docx or xml file.

It is possible to implement additional export modules by implementing IExportModule.

![exportcollapsed](https://cloud.githubusercontent.com/assets/18663475/18163495/954d46e8-703b-11e6-8c0a-24a7729af282.png)
![exportexpanded](https://cloud.githubusercontent.com/assets/18663475/18163501/9910773c-703b-11e6-8695-d6b89d068d2a.png)

petrsvihlik commented on KInspector.Web/WebAPI/Controllers/ExportController.cs:
```diff
@@ -0,0 +1,100 @@
+
+                    // Send stream
+                    var response = Request.CreateResponse(HttpStatusCode.OK);
+                    response.Content = new ByteArrayContent(memoryStream.ToArray());
```
why not to use `StreamContent` to prevent copying byte arrays back and forth...?

petrsvihlik commented on KInspector.Modules/Export/Modules/ExportXlsx.cs:
```diff
@@ -0,0 +1,75 @@
+                }
+            }
+
+            MemoryStream stream = new MemoryStream();
```
http://stackoverflow.com/a/37398007/1332034
-> this could reduce the memory impact...I'd go for it

petrsvihlik commented:
discovered while testing:
- when I export to word the first page is blank (while it should contain "Kinspector report"
- can you make it a little bit more user friendly?
 - use "Excel/Word" instead of xlsx/docx in the dropdown
 - is it necessary to have the module enabled in the "Setup" section?
 - module selection - is it wise to select all modules by default? uneducated user may be surprised why it takes so long to export... I'd probably invert it and implement a "(de)select all" functionality in another PR

code review:
- the performance related comments in the code (they apply to all export modules)
- all public members have to be commented using xmldoc (especially things like interfaces, extension methods) + go through the export implementations and add comments where it would make sense

otherwise it looks and works reeealy great! good job @JosefDvorak !

JosefDvorak commented:
- I've fixed the docx template. The NPOI library seems to have problem with placing text in a text section.
- I've updated the naming to Word/Excel.
- I was thinking about hiding the Export button in setup module as well, but then I figured there is no real harm in having it there. I won't break anything, just export results of the setup scripts. I'd leave it up to the user if they wants to use the export functionality with these modules.
- I pre-selected the modules because I thought that would be the more user friendly way. I'm ok with having them un-selected by default, so I removed the pre-selection.

- I've implemented the lock-able MemoryStream workaround
- I've tried to use the StreamContent before, but kept running into issues where the file size was 0. I've found the issue and made it work.

- Last thing to finish are the comments.

petrsvihlik commented:
good job, @JosefDvorak! thank you :)
